src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -297,28 +293,48 @@ private constructor(
 
     @RequiresEdt
     fun createFromState(project: Project, state: ChatState): AgentChatSession {
-      val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val modelFromState = state.model?.let { ChatModel.fromDisplayName(it) }
-      val chatSession = AgentChatSession(project, sessionId, state.internalId!!, modelFromState)
-      state.messages.forEachIndexed { index, message ->
-        val parsed =
-            when (val speaker = message.speaker) {
-              MessageState.SpeakerState.HUMAN -> Speaker.HUMAN
-              MessageState.SpeakerState.ASSISTANT -> Speaker.ASSISTANT
-              else -> error("unrecognized speaker $speaker")
+      val agentChatSession = CompletableFuture<AgentChatSession>()
+      createNewPanel(project) { codyAgent ->
+        val innerSessionId = codyAgent.server.chatNew()
+        val modelFromState =
+            state.llm?.let {
+              ChatModelsResponse.ChatModelProvider(
+                  default = false,
+                  codyProOnly = false,
+                  provider = it.provider ?: "Unknown",
+                  title = it.title ?: "Unknown",
+                  model = it.model ?: "")
```

Comment:
I guess it's just a safeguard as we expect ot always get it, right?
Just to be extra safe we could also do:
model = it.model ?: it.provider ?: "Default"